### PR TITLE
:ambulance: fix redirection and mandatory fields

### PR DESCRIPTION
**Pull request by trojikman**

### CODE REVIEW

File: website_sale_checkout_store/controllers/main.py [hunk 1]
```diff
@@ -52,20 +52,23 @@ def payment_get_status(self, sale_order_id, **post):
...
-                return ["name", "phone", "email"]
+                fields = ["name", "phone"]
             else:
-                return ["name", "phone", "email", "country_id"]
+                fields = ["name", "phone", "country_id"]
         else:
-            return ["name", "phone", "email", "street", "city"]
+            fields = ["name", "phone", "street", "city"]
+        if not shipping:
+            fields.append('email')
```
Comment (yelizariev) [inline website_sale_checkout_store/controllers/main.py]: Does that mean, that for shipping case email is not required? Why? Does the module docs says that or opposite or nothing about it?

Comment (trojikman) [reply website_sale_checkout_store/controllers/main.py]: This is stated in the original method `_get_mandatory_shipping_fields` https://github.com/odoo/odoo/blob/12.0/addons/website_sale/controllers/main.py#L525

Comment (trojikman) [reply website_sale_checkout_store/controllers/main.py]: in the case when a new address is added, this method is called but there is no e-mail field in the form, which causes an error

Comment (yelizariev) [reply website_sale_checkout_store/controllers/main.py]: ok

**Pull request merged**